### PR TITLE
added recipe for karchive

### PR DESCRIPTION
**Pull request by boussaffawalid**
Specify library name and version:  **karchive/5.75.0**
---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/karchive/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        "with_lzma": [True, False],
+        "fPIC": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_lzma": True
+    }
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake"
```
Comment (SSE4) [inline recipes/karchive/all/conanfile.py:26]:
```suggestion
    generators = "cmake", "cmake_find_package_multi"
```
seems like it uses `find_package` to find `qt5`

Comment (boussaffawalid) [reply recipes/karchive/all/conanfile.py:26]: @SSE4 I was trying to use cmakedeps with cmaketoolchain but I was not able to get it to work. I will give it a shot with your suggestion

File: recipes/karchive/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    def requirements(self):
+        self.requires("qt/5.15.2")
+        self.requires("zlib/1.2.11")
+        self.requires("bzip2/1.0.8")
+        if self.options.with_lzma:
+            self.requires("xz_utils/5.2.5")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        rename(self, "karchive-{}".format(self.version), self._source_subfolder)
```
Comment (Croydon) [inline recipes/karchive/all/conanfile.py:50]: We can do that smarter by now:

```suggestion
        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
```

File: recipes/karchive/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
+import os
+from conan.tools.files import rename
```
Comment (Croydon) [inline recipes/karchive/all/conanfile.py:2]:
```suggestion
```

File: recipes/karchive/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+from conans import ConanFile, CMake, tools
+import shutil
+
+
```
Comment (Croydon) [inline recipes/karchive/all/conanfile.py:6]: Please always add this attribute

```suggestion
required_conan_version = ">=1.33.0"


```

File: recipes/karchive/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+        self.copy("*", src=os.path.join(self._source_subfolder, "LICENSES"), dst="licenses")
+
+        shutil.rmtree(os.path.join(self.package_folder, "lib", "cmake"))
```
Comment (Croydon) [inline recipes/karchive/all/conanfile.py:71]:
```suggestion
        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
```

Use Conan methods when available

File: recipes/karchive/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,79 @@
+import os
+from conan.tools.files import rename
+from conans import ConanFile, CMake, tools
+import shutil
```
Comment (Croydon) [inline recipes/karchive/all/conanfile.py:4]:
```suggestion
```

File: recipes/karchive/all/conanfile.py (hunk 2 above)
Comment (boussaffawalid) [reply recipes/karchive/all/conanfile.py:50]: that is a good one ;)

File: recipes/karchive/all/test_package/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+from conan.tools.cmake import CMake, CMakeToolchain
+from conan.tools.build import cross_building as tools_cross_building
+from conan.tools.layout import cmake_layout
+
+required_conan_version = ">=1.43.0"
+
+
+class TestPackageConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "CMakeDeps"
```
Comment (SSE4) [inline recipes/karchive/all/test_package/conanfile.py:12]:
```suggestion
    generators = "CMakeDeps", "cmake"
```

**Pull request closed without merging**